Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -140,13 +140,16 @@ void MediaPlayerPrivateGStreamerMSE::play()
 {
     GST_DEBUG_OBJECT(pipeline(), "Play requested");
     m_isPaused = false;
+    if (!m_playbackRate && m_playbackRatePausedState == PlaybackRatePausedState::ManuallyPaused)
```

Comment:
Given the case of `m_playbackRate == 0` when `play()` is called, can we be in `PlaybackRatePausedState::Playing`? If so, how? If not, as I would assume, that would make for a good assert.

What other states can we be in here? If they are `ManuallyPaused` and `RatePaused`, both would do the same outcome, making the second part of the `if` redundant.

Reply:
In the first case, it can't happen. I guess you mean we should assert before the if.

The other states can happen but not all of them with `m_playbackRate == 0`.